services/api/src/typeDefs.js

```diff
@@ -461,7 +465,7 @@ const typeDefs = gql`
     """
     Reference to OpenShift Object this Project should be deployed to
     """
-    openshift: Openshift
+    openshift: Kubernetes
```

Comment:
we also discussed that we want the possibility to define multiple deploytargets based on the type of environment. So you can define a cluster for production and one for development environment types.
this also means we need to store the actual deployed cluster on the environment itself.

Reply:
It was a bit tricky to come up with a solution for this but added a new `DeploymentTargetRule` type that works similar to billing modifiers that I think can handle all the situations that could come up.